src/clp_logging/readers.py

```diff
@@ -387,3 +388,251 @@ def __init__(
     def dump(self) -> None:
         for log in self:
             stderr.write(log.formatted_msg)
+
+
+class CLPSegmentStreamingReader:
+    """
+    Abstract reader class used to read stream segments for CLP IR/"logs"
```

Comment:
Edit: This class is instantiated at line 631, so maybe it is private rather than abstract?

If this class is abstract we should be using python's abstract base class (ABC) library:
e.g. `CLPSegmentStreamingReader(metaclass=ABCMeta):`
see: https://docs.python.org/3/library/abc.html

Reply:
True. Made it private now.